src/orion/core/worker/trial.py

```diff
@@ -155,7 +155,6 @@ def __init__(self, **kwargs):
                 setattr(self, attrname, None)
 
         self.status = 'new'
-        self._id = self.NoID
```

Comment:
Revert this please. Default trial ID is `Trial.NoID`, which in current agreed semantics is Python's `None`, but in the future it could be something else.

Reply:
A trial would now have an id by construction, it cannot be Trial.NoID anymore.

Reply:
If we guarantee that we can create an id during initialization, then should also `assert params in kwargs and experiment in kwargs`.

Reply:
In my opinion, a non-configured Trial should raise a RuntimeError whenever we try to hash it. Because of this, we don't need to test for params and experiment during initialization.

Reply:
nevermind on that